extensions/gitpod/src/remoteConnector.ts

```diff
@@ -403,40 +481,170 @@ export default class LocalApp extends Disposable {
+		const forceUseLocalApp = vscode.workspace.getConfiguration('gitpod').get<boolean>('remote.useLocalApp')!;
+
+		const params: SSHConnectionParams = JSON.parse(uri.query);
+		if (new URL(params.gitpodHost).host !== new URL(gitpodHost).host) {
+			const yes = 'Yes';
+			const cancel = 'Cancel';
+			const action = await vscode.window.showInformationMessage(`Trying to connect to a remote workspace in a different Gitpod Host. Continue and update 'gitpod.host' setting to '${params.gitpodHost}'?`, yes, cancel);
+			if (action === cancel) {
+				return;
+			}
+
+			await vscode.workspace.getConfiguration('gitpod').update('host', params.gitpodHost, vscode.ConfigurationTarget.Global);
+			this.logger.info(`Updated 'gitpod.host' setting to '${params.gitpodHost}' while trying to connect to a remote workspace`);
+		}
+
+		this.logger.info('Opening remote workspace', uri.toString());
+
+		let sshDestination: string | undefined;
+		if (!forceUseLocalApp) {
+			try {
+				this.telemetry.sendTelemetryEvent('gitpod_desktop_ssh', { kind: 'gateway', status: 'connecting' });
+
+				sshDestination = await this.getWorkspaceSSHDestination(params.workspaceId, params.gitpodHost);
+
+				this.telemetry.sendTelemetryEvent('gitpod_desktop_ssh', { kind: 'gateway', status: 'connected' });
+			} catch (e) {
+				if (e instanceof Error && e.message === 'no_ssh_gateway') {
+					this.logger.error(`SSH gateway not configured for this Gitpod Host ${params.gitpodHost}`);
+					vscode.window.showWarningMessage(`${params.gitpodHost} does not support [direct SSH access](https://github.com/gitpod-io/gitpod/blob/main/install/installer/docs/workspace-ssh-access.md), connecting via the deprecated SSH tunnel over WebSocket.`);
+					this.telemetry.sendTelemetryEvent('gitpod_desktop_ssh', { kind: 'gateway', status: 'failed', reason: 'no-ssh-gateway' });
+					// Do nothing and continue execution
+				} else if (e instanceof Error && e.message === 'no_running_instance') {
+					this.logger.error(`No running instance for this workspaceId ${params.workspaceId}`);
+					vscode.window.showErrorMessage(`Failed to connect to remote workspace: No running instance for '${params.workspaceId}'`);
+					this.telemetry.sendTelemetryEvent('gitpod_desktop_ssh', { kind: 'gateway', status: 'failed', reason: 'ws-not-running' });
+					return;
+				} else {
+					if (e instanceof SSHError) {
+						this.logger.error('SSH test connection error', e.cause);
+						this.telemetry.sendTelemetryEvent('gitpod_desktop_ssh', { kind: 'gateway', status: 'failed', reason: 'ssh-blocked' });
+					} else {
+						this.logger.error(`Failed to connect to remote workspace ${params.workspaceId}`, e);
+						this.telemetry.sendTelemetryEvent('gitpod_desktop_ssh', { kind: 'gateway', status: 'failed', reason: 'other-error' });
+					}
+					const seeLogs = 'See Logs';
+					const action = await vscode.window.showErrorMessage(`Failed to connect to remote workspace ${params.workspaceId}`, seeLogs);
```

Comment:
We need to add SSH docs in VS Code Desktop action which explains 2 different modes, pitfall and workarounds to switch between them, as well points to general troubleshooting docs.

Here we need a new action like `Show Troubleshooting` or something like that which takes a user to such docs.

Please add such action also for local app case.

Reply:
PR updating docs https://github.com/gitpod-io/website/pull/2085